openapi/connector_mgmt.yaml

```diff
@@ -1175,6 +1175,7 @@ components:
         - deleted
         - provisioning
         - deprovisioning
+        - missing-secrets
```

Comment:
Is it correct to have `missing-secrets` as part of the public  APIs ?
nitpick, it should use snake case `missing_secrrets`

Reply:
In my opinion no, it should not leak into the api. It is more  like an implementation detail.

Reply:
As a user, why wouldn't you want to know that there is something wrong with your connector? What's the argument for hiding this condition from the user?

Reply:
IMHO, I don't think this is a phase is more an error condition

Reply:
Ok, so when the agent sees this, will it respond with a `failed` phase and an error message in a status condition? I just want to make sure it's visible to the user in case it happens when the agent needs the spec for starting/updating. 
If that's the case, the fleetmanager can just pass this value along without updating the DB expecting the agent to circle back with an error. 

BTW, if the value is not part of the enum, won't it break API generated clients that might be validating the string?

Reply:
> Ok, so when the agent sees this, will it respond with a `failed` phase and an error message in a status condition? I just want to make sure it's visible to the user in case it happens when the agent needs the spec for starting/updating. If that's the case, the fleetmanager can just pass this value along without updating the DB, expecting the agent to circle back with an error.
> 

Yes so, we can think about something like:
- if transitioning the connector to a stop/delete phase, then the presence of secrets may not be relevant hence, the action can be performed.
- if reify is required, then the process will fail and an error will be reported

In bot cases, we can add a condition that signal the lack of spec, like:

```yaml
- type: ConnectorSpecAvailable 
  status: "False"
  reason: "ConnectorSpecNotAvailable"
  message: "..."
```
And in case of failure, we can use the `Ready` condition to report the error.

> BTW, if the value is not part of the enum, won't it break API generated clients that might be validating the string?

If `missing-secrets` is not a phase, then I don't think we have such issue as the user should use the `failed` phase and eventually check the error.

Reply:
```
If missing-secrets is not a phase, then I don't think we have such issue as the user should use the failed phase and eventually check the error.
```
Deployment status phase references `ConnectorState` enum. Is the client in the agent using Java enums or String for it?

Reply:
enum

Reply:
If we don't want to treat `secrets_unavailable` as a phase, we should go back to what I did at first for this PR, i.e. make it a separate flag. That way we don't pollute the phase enum with a value that's used to convey something else. 

So, in case secrets are not available:
* connector spec will be empty
* deployment will include a flag `secrets_unavailable=true`

Reply:
Added required flag `resolved_secrets` as discussed.